app/src/main/java/org/schabi/newpipe/player/Player.java

```diff
@@ -1682,42 +1696,79 @@ public void showControlsThenHide() {
         if (DEBUG) {
             Log.d(TAG, "showControlsThenHide() called");
         }
-        showOrHideButtons();
-        showSystemUIPartially();
-
         final int hideTime = binding.playbackControlRoot.isInTouchMode()
-                ? DEFAULT_CONTROLS_HIDE_TIME
-                : DPAD_CONTROLS_HIDE_TIME;
-
-        showHideShadow(true, DEFAULT_CONTROLS_DURATION);
-        animate(binding.playbackControlRoot, true, DEFAULT_CONTROLS_DURATION,
-                AnimationType.ALPHA, 0, () -> hideControls(DEFAULT_CONTROLS_DURATION, hideTime));
+            ? DEFAULT_CONTROLS_HIDE_TIME
+            : DPAD_CONTROLS_HIDE_TIME;
+        showControls(
+            DEFAULT_CONTROLS_DURATION,
+            () -> hideControls(DEFAULT_CONTROLS_DURATION, hideTime)
+        );
     }
 
     public void showControls(final long duration) {
+        showControls(duration, null);
+    }
+
+    public void showControls(final long duration, final Runnable callback) {
         if (DEBUG) {
             Log.d(TAG, "showControls() called");
         }
-        showOrHideButtons();
-        showSystemUIPartially();
+
+        final RelativeLayout layoutToShow;
+        if (!isLocked) {
+            showOrHideButtons();
+            showSystemUIPartially();
+            layoutToShow = binding.playbackControlRoot;
+        } else {
+            layoutToShow = binding.lockedRoot;
+        }
+
         controlsVisibilityHandler.removeCallbacksAndMessages(null);
         showHideShadow(true, duration);
-        animate(binding.playbackControlRoot, true, duration);
+        animate(
+            layoutToShow,
+            true,
+            duration,
+            AnimationType.ALPHA,
+            0,
+            callback
+        );
     }
 
     public void hideControls(final long duration, final long delay) {
+        hideControls(duration, delay, null);
+    }
+
+    public void hideControls(final long duration, final long delay, final Runnable callback) {
         if (DEBUG) {
             Log.d(TAG, "hideControls() called with: duration = [" + duration
                     + "], delay = [" + delay + "]");
         }
 
-        showOrHideButtons();
+        final RelativeLayout layoutToHide;
+        if (!isLocked) {
+            showOrHideButtons();
+            layoutToHide = binding.playbackControlRoot;
+        } else {
+            layoutToHide = binding.lockedRoot;
+        }
```

Comment:
Instead of resolving which layout to hide/show, is it not possible to group them together and show/hide the common parent?

Reply:
Either way the logic would fork to show one sub-layout or the other, so I don't think there's much benefit.

Reply:
Does the visibility of a child component override the one the parent component?

What i imaged is that tapping the screen to show the controls, toggles the visibility of the common parent component. Which hides both children, no matter what visibility they have. pressing the lock/unlock toggles the visibility of both child components, so the other one is shown